src/module/server-group/endpoints/config/EndpointConfigCtrl.ts

```diff
@@ -0,0 +1,114 @@
+import {IStateService} from "angular-ui-router";
+import {AbstractConfigurationCtrl} from "../../../../common/configuration/AbstractConfigurationCtrl";
+import IModalService = angular.ui.bootstrap.IModalService;
+import {LaunchTypeService} from "../../../../services/launchtype/LaunchTypeService";
+import {ServerGroupService} from "../../../../services/server-group/ServerGroupService";
+import {IServerGroup} from "../../../../services/server-group/IServerGroup";
+import {IEndpoint} from "../../../../services/endpoint/IEndpoint";
+import {openConfirmationModal, openErrorModal, openRestartModal} from "../../../../common/dialogs/Modals";
+import {CompositeOpBuilder} from "../../../../services/dmr/CompositeOpBuilder";
+import {EndpointService} from "../../../../services/endpoint/EndpointService";
+
+export class EndpointConfigCtrl extends AbstractConfigurationCtrl {
+  static $inject: string[] = ["$state", "$scope", "$uibModal", "serverGroupService", "endpointService", "launchType",
+    "serverGroup", "endpoint", "endpointMeta", "endpointType", "endpointName"];
+
+  readOnlyFields: string[];
+
+  private typeChangeCancelled: boolean = false;
+
+  constructor(private $state: IStateService,
+              private $scope: ng.IScope,
+              private $uibModal: IModalService,
+              private serverGroupService: ServerGroupService,
+              private endpointService: EndpointService,
+              private launchType: LaunchTypeService,
+              private serverGroup: IServerGroup,
+              private endpoint: IEndpoint,
+              private endpointMeta: any,
+              private endpointType: string,
+              private endpointName: string) {
+    super();
+    console.log(endpoint);
+  }
+
+  goToContainerCachesView(): void {
+    this.$state.go("container.caches", {
+      //profileName: this.profile,
+      //containerName: this.containerName
+    });
+  }
+
+  goToEndpointsView(): void {
+    this.$state.go("server-group.endpoints", {serverGroup: this.serverGroup.name});
+  }
+
+  create(endpoint: IEndpoint): ng.IPromise<any> {
+    let excludedAttributes: string [] = ["is-new-node"];
+    return this.endpointService.save(endpoint, excludedAttributes);
+  }
+
+  update(endpoint: IEndpoint): ng.IPromise<any> {
+    let excludedAttributes: string [] = ["socket-binding", "is-new-node"];
+    return this.endpointService.save(endpoint, excludedAttributes);
+  }
```

Comment:
Yes, but we need different exclude exclusion lists for save and update. I thought of this way as the best approach to differentiate the two. EndpointsService should not care about specific exclusion lists, higher API that calls it - should. Does that make more sense?